library/src/main/java/com/google/android/exoplayer/util/CodecSpecificDataUtil.java

```diff
@@ -49,13 +49,22 @@ private CodecSpecificDataUtil() {}
    */
   public static Pair<Integer, Integer> parseAudioSpecificConfig(byte[] audioSpecificConfig) {
     int audioObjectType = (audioSpecificConfig[0] >> 3) & 0x1F;
-    int byteOffset = audioObjectType == 5 || audioObjectType == 29 ? 1 : 0;
-    int frequencyIndex = (audioSpecificConfig[byteOffset] & 0x7) << 1
-        | ((audioSpecificConfig[byteOffset + 1] >> 7) & 0x1);
-    Assertions.checkState(frequencyIndex < 13);
-    int sampleRate = AUDIO_SPECIFIC_CONFIG_SAMPLING_RATE_TABLE[frequencyIndex];
-    int channelCount = (audioSpecificConfig[byteOffset + 1] >> 3) & 0xF;
-    return Pair.create(sampleRate, channelCount);
+    if (audioObjectType != 31) {
```

Comment:
Make it (audioObjectType < 31) just to be consistent with buildAudioSpecificConfig().